test/jdk/java/security/KeyStore/CheckMacOSKeyChainIntermediateCATrust.java

```diff
@@ -0,0 +1,169 @@
+    }
+
+    @BeforeEach
+    public void setup() {
+        System.out.println("Adding certificates to key chain");
+        addCertificatesToKeyChain();
+    }
+
+    @AfterEach
+    public void cleanup() {
+        System.out.println("Cleaning up");
+        deleteCertificatesFromKeyChain();
+    }
+
+    private static void addCertificatesToKeyChain() {
+        String loginKeyChain = getLoginKeyChain();
+
+        Path caPath = Path.of("%s/%s".formatted(DIR, "test-ca.pem"));
+        List<String> args = List.of(
+                "/usr/bin/security",
+                "add-trusted-cert",
+                "-k", loginKeyChain,
+                caPath.toString()
+        );
+        executeProcess(args);
+
+        caPath = Path.of("%s/%s".formatted(DIR, "test-intermediate-ca.pem"));
+        args = List.of(
+                "/usr/bin/security",
+                "add-certificates",
+                "-k", loginKeyChain,
+                caPath.toString()
+        );
+        executeProcess(args);
+
+    }
+
+    private static String getLoginKeyChain() {
+        return Path.of(System.getProperty("user.home"), "Library/Keychains/login.keychain-db").toString();
+    }
+
+    private static void executeProcess(List<String> params) {
+        System.out.println("Command line: " + params);
+        try {
+            int exitStatus = ProcessTools.executeProcess(params.toArray(new String[0])).getExitValue();
+            if (exitStatus != 0) {
+                fail("Process started with: " + params + " failed");
+            }
+        } catch (Throwable e) {
+            fail(e.getMessage());
+        }
+    }
+
+    private static void deleteCertificatesFromKeyChain() {
+        executeProcess(
+                List.of(
+                        "/usr/bin/security",
+                        "delete-certificate",
+                        "-c", "Example CA",
+                        "-t"
+                )
+        );
+
+        executeProcess(
+                List.of(
+                        "/usr/bin/security",
+                        "delete-certificate",
+                        "-c", "Example Intermediate CA",
+                        "-t"
+                )
+        );
+    }
+
+    private static boolean containsSubjectName(List<X509Certificate> certificates, String subjectName) {
+        return certificates.stream()
+                .map(cert -> cert.getSubjectX500Principal().getName())
+                .anyMatch(name -> name.contains(subjectName));
+    }
+
+    private static List<String> getSubjects(List<X509Certificate> certificates) {
+        return certificates.stream()
+                .map(cert -> cert.getSubjectX500Principal().getName())
+                .toList();
+    }
+
+    private static void assertThat(boolean expected, String message, List<X509Certificate> certificates) {
+        if (!expected) {
+            throw new AssertionError(message + ", subjects: " + getSubjects(certificates));
```

Comment:
I do not like printing all KeyChain certificates on the failure. It could be sensitive information.

If you do not collect all certificates, the test could be simplified - without Stream API

Reply:
Without this its harder to debug what went wrong, this tells you what subjects are found, its not the certificate itself, just e.g. 

```
java.lang.AssertionError: Non trusted CA not found CN=Non Trusted Example CA,O=Example,C=US, subjects: [ CN=StartCom Certification Authority,OU=Secure Digital Certificate Signing,O=StartCom Ltd.,C=IL, CN=TIMJA-INTERMEDIATE,O=TIMJA,ST=ES,C=UK, CN=TIMJA-INTERMEDIATE-2,O=TIMJA,ST=ES,C=UK, CN=TIMJA-ROOT,O=TIMJA,ST=ES]
```

Reply:
I understand the reason for the test development, but I am still concerned about printing this information.
Is it possible it will be disabled by default but could be enabled from the cmdline or changing the source code?
`
    static boolean verbose = false; // avoid too verbose output
`